src/cdk/tree/node.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {FocusableOption} from '@angular/cdk/a11y';
+import {
+  Directive,
+  ElementRef,
+  Input,
+  OnDestroy,
+  TemplateRef
+} from '@angular/core';
+import {takeUntil} from 'rxjs/operator/takeUntil';
+import {Subject} from 'rxjs/Subject';
+import {CdkTree} from './tree';
+import {getTreeControlFunctionsMissingError} from './tree-errors';
+
+
+/** Context provided to the ndoes */
+export class CdkTreeNodeOutletContext<T> {
+
+  static mostRecentContextData: any;
+
+  /** Data for the node. */
+  $implicit: T;
+
+  /** Index location of the node. */
+  index?: number;
+
+  /** Length of the number of total nodes. */
+  count?: number;
+
+  constructor(data: T) {
+    this.$implicit = data;
+    CdkTreeNodeOutletContext.mostRecentContextData = data;
+  }
+}
+
+/**
+ * Data node definition for the CdkTree.
+ * Captures the node's template and a when predicate that describes when this node should be used.
+ */
+@Directive({
+  selector: '[cdkTreeNodeDef]',
+  inputs: [
+    'when: cdkTreeNodeDefWhen'
+  ],
+})
+export class CdkTreeNodeDef<T> {
+  /**
+   * Function that should return true if this node template should be used for the provided node
+   * data and index. If left undefined, this node will be considered the default node template to
+   * use when no other when functions return true for the data.
+   * For every node, there must be at least one when function that passes or an undefined to
+   * default.
+   */
+  when: (nodeData: T, index: number) => boolean;
+
+  /** @docs-private */
```

Comment:
You should be able to set this on the directive doc, rather than the constructor.